src/make-styles.tsx

```diff
@@ -203,15 +215,31 @@ export const makeTemplateFunction = <
     const styles = createNestedStyleObject(cssString);
     StyledForwardRefRenderFunction = (
       // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-      // @ts-ignore ts doesn't understand that A is not allwoed to declar children and style
+      // @ts-ignore ts doesn't understand that A is not allowed to declare children and style
       { children, style, ...props }: Omit<P & I, RequiredKeys<A>>,
       ref
     ) => {
       const theme = useTheme();
       const dimensions = useWindowDimensions();
-      let styleProps: { [key: string]: Style | Style[] } = useStyleSheet(styles, theme, dimensions);
+      const componentDimensions = useContainerDimensions();
+
+      //if no container is provided, @container query uses window dimensions
+      const containerDimensions = useContext(ContainerSizeContext) ?? dimensions;
+      const isContainer = Object.values(styles).some((s) => s.main.contain || s.main.containerType || s.main.containerName || s.main.container)
+      
+      let styleProps: { [key: string]: Style | Style[] } = useStyleSheet(styles, theme, dimensions, containerDimensions);
       styleProps = style ? { ...styleProps, style: [styleProps.style, style] } : styleProps;
       const transformedProps = transformProps({ ...props, theme } as AttrProps<P, I, A>);
+
+      if(isContainer) {
+        return (
+          <ContainerSizeContext.Provider value = {componentDimensions[0]}>
+            <Component {...filterComponentProps(transformedProps)} {...styleProps} ref={ref} onLayout={componentDimensions[1]}>
+              {children}
```

Comment:
I **think** the `ContainerSizeContext.Provider` needs to go on the inside of the component not on the outside. Maybe it doesn't matter just wondering what happens if the container is using a container query itself.